infra/build_specified_commit.py

```diff
@@ -54,42 +59,116 @@ def build_fuzzer_from_commit(project_name,
+      source_path=old_repo_manager.repo_dir,
+      mount_location=os.path.join('/src', old_repo_manager.repo_name))
+
+
+def run_command_in_image(image_name, command):
+  """Runs a specific command inside a docker image and returns the results.
+
+  Args:
+    image_name: The docker image for the command to be run in
+    command: The command to be run in the image
+
+  Returns:
+    the output of the command
+
+  Raises:
+    DockerExecutionError: on commands execution failing
+  """
+  command_to_run = [
+      'docker', 'run', '--rm', '-i', '--privileged', '-t', image_name
+  ]
+  command_to_run.extend(command)
+  print('Running command: %s' % command_to_run)
+  process = subprocess.Popen(command_to_run, stdout=subprocess.PIPE)
+  out, err = process.communicate()
+  if err:
+    raise DockerExecutionError('Error running command: %s with error: %s' %
+                               (command_to_run, err.decode('ascii')))
+  if out:
+    return out.decode('ascii'), process.returncode
+  return None, process.returncode
+
+
+def check_docker_for_commit(docker_image_name, dir_name, example_commit):
+  """ Checks a docker image directory for a specific commit.
+
+  Args:
+    docker_image_name: The name of the projects docker image
+    dir_name: The name of the directory to test for the commit
+    example_commit: The commit SHA to check for
 
+  Returns:
+    True if docker image directory contains that commit
+  """
+  dir_to_check = '/src/' + dir_name
+
+  #Check if valid git repo
+  out, returncode = run_command_in_image(
+      docker_image_name, ['/bin/bash', '-c', 'ls -a ' + dir_to_check])
+  if '.git' not in out:
+    return False
+
+  #Check if history fetch is needed
+  check_shallow_command = '[ -f ' + dir_to_check + '/.git/shallow ]'
+  out, returncode = run_command_in_image(
+      docker_image_name, ['/bin/bash', '-c', check_shallow_command])
+
+  if returncode == 0:
+    #Check if commit exists
+    _, returncode = run_command_in_image(docker_image_name, [
+        '/bin/bash', '-c', 'cd ' + dir_to_check +
+        '; git fetch --unshallow; git cat-file -e ' + example_commit
+    ])
+  else:
+    _, returncode = run_command_in_image(docker_image_name, [
+        '/bin/bash', '-c',
+        'cd ' + dir_to_check + '; git cat-file -e ' + example_commit
+    ])
+  if returncode == 0:
+    return True
+  return False
+
+
+def get_repo_url(docker_image_name, dir_name):
+  """Gets a git repo URL from a specific directory in a docker image.
 
-def infer_main_repo(project_name, local_store_path, example_commit=None):
+  Args:
+    docker_image_name: The name of the image where the git repo is located
+    dir_name: The directory on the image where the git repo exists
+
+  Returns:
+    The repo URL string
+  """
+  command_to_run = 'cd /src/' + dir_name + '; git config --get remote.origin.url'
+  out, returncode = run_command_in_image(docker_image_name,
+                                         ['/bin/bash', '-c', command_to_run])
```

Comment:
nit: just `git -C <src-dir> command` can make a lot of these git commands easier. You may even be able to get rid of bash in some cases.

Reply:
+1